src/index.rs

```diff
@@ -1133,4 +1133,225 @@ mod tests {
+                let mut file_name = String::from(FILE_NAME);
+                file_name.push_str(&i.to_string());
+
+                let mut file_path = path.clone();
+                let mut new_file_path = path.clone();
+                file_path.push(file_name.clone());
+
+                let rand_new_filename: String = new_rand_filename();
+                new_file_path.push(rand_new_filename.clone());
+
+                let old_id: ResourceId;
+                match CanonicalPathBuf::canonicalize(&file_path) {
+                    Ok(canonicalized_path) => {
+                        old_id = initial_index.path2id
+                            [&canonicalized_path.clone()]
+                            .id;
+                    }
+                    Err(_) => {
+                        log::warn!(
+                            "File {} not found",
+                            file_path.to_str().unwrap_or("no_path_buf")
+                        );
+                        continue;
+                    }
+                }
+
+                let update_state = generate_random_update(
+                    path.clone(),
+                    Some(&file_name),
+                    Some(&rand_new_filename),
+                );
+
+                match update_state.try_into() {
+                    Ok(FileAction::CREATE) => {
+                        let _ = index1.track_addition(&new_file_path);
+                    }
+                    Ok(FileAction::UPDATE) => {
+                        let _ = index1.track_update(&file_path, old_id);
+                    }
+                    Ok(FileAction::DELETE) => {
+                        let _ = index1.track_deletion(old_id);
+                    }
+                    Ok(FileAction::MOVE) => {
+                        let _ = index1.track_deletion(old_id);
+                        let _ = index1.track_addition(&new_file_path);
+                    }
+                    Err(_) => println!("rnd_num error"),
+                }
+            }
+            index2
+                .update_all()
+                .expect("Should update index correctly");
+            assert_eq!(index1, index2);
+        })
+    }
+
+    #[test]
+    fn update_all_compare_track_addition() {
+        run_test_and_clean_up(|path| {
+            create_file_at(path.clone(), Some(DATA_SIZE_1), Some(FILE_NAME_1));
+
+            let initial_index = ResourceIndex::build(path.clone());
+            let mut index_track_addition = initial_index.clone();
+            let mut index_update_all = initial_index.clone();
+
+            create_file_at(path.clone(), Some(DATA_SIZE_1), Some(FILE_NAME_2));
+
+            let mut added_file_path = path.clone();
+            added_file_path.push(FILE_NAME_2);
+            let _ = index_track_addition.track_addition(&added_file_path);
+
+            index_update_all
+                .update_all()
+                .expect("Should update index correctly");
+
+            assert_ne!(
```

Comment:
After this assertion we should `sleep` for 1ms, perform `update_all` again and `assert_eq` new index.

Reply:
regarding this test,
Why should the result from `track_addition` and `update_all` be different?